modules/Movecraft/src/main/java/net/countercraft/movecraft/warfare/assault/AssaultTask.java

```diff
@@ -41,52 +41,66 @@ public void run() {
             //first, find a position for the repair beacon
             int beaconX = assault.getMinPos().getBlockX();
             int beaconZ = assault.getMinPos().getBlockZ();
-            int beaconY = w.getHighestBlockAt(beaconX, beaconZ).getY();
-            int x, y, z;
-            for (x = beaconX; x < beaconX + 5; x++)
-                for (z = beaconZ; z < beaconZ + 5; z++)
-                    if (!w.isChunkLoaded(x >> 4, z >> z))
-                        w.loadChunk(x >> 4, z >> 4);
-            boolean empty = false;
-            while (!empty && beaconY < 250) {
-                empty = true;
-                beaconY++;
-                for (x = beaconX; x < beaconX + 5; x++) {
-                    for (y = beaconY; y < beaconY + 4; y++) {
-                        for (z = beaconZ; z < beaconZ + 5; z++) {
-                            if (!w.getBlockAt(x, y, z).isEmpty())
-                                empty = false;
+            boolean good = false;
+            int beaconY = 0;
+            for(int i = 0; i < 255; i++) {
+                if(w.getBlockAt(beaconX, i, beaconZ).getType() == Material.AIR) {
+                    good = true;
+                    beaconY = i;
+                    break;
+                }
+            }
+            if(!good) {
+                Bukkit.getServer().broadcastMessage(String.format("BEACON PLACEMENT FOR %s FAILED, CONTACT AN ADMIN!", assault));
+            }
```

Comment:
```java
for(beaconY = 255; beaconY >0; beaconY--){
    if(w.getBlockAt(beaconX, i, beaconZ).getType() != Material.AIR) {
        beacony++;
        break;
    }
}
if(y>255){
    Bukkit.getServer().broadcastMessage(String.format("BEACON PLACEMENT FOR %s FAILED, CONTACT AN ADMIN!", assault));
}

```
This will cause the beacon to be placed as close to the highest possible surface